general-superstaq/general_superstaq/superstaq_client.py

```diff
@@ -920,6 +933,53 @@ def __repr__(self) -> str:
         )
 
 
+def read_ibm_credentials(ibm_name: str) -> dict[str, str]:
+    """Function to try to read IBM credentials from .qiskit/qiskit-ibm.json.
+
+    Args:
+        ibm_name: The name under which the IBM account credentials are locally stored.
+
+    Raises:
+        FileNotFoundError: If the file is not found.
+        KeyError: If the provided `ibm_name` does not have credentials stored in the
+            config file or `token` and/or `channel` keys are missing for the credentials
+            of the account under `ibm_name`.
+
+    Returns:
+        Dictionary containing the ibm token, channel, and instance (if available).
+    """
+    data_dir = pathlib.Path(os.getenv("XDG_DATA_HOME", "~/.local/share")).expanduser()
+    home_dir = pathlib.Path.home()
+    for directory in [
+        home_dir.joinpath(".qiskit"),
+        data_dir.joinpath("super.tech"),
+        data_dir.joinpath("coldquanta"),
+        home_dir.joinpath(".super.tech"),
+        home_dir.joinpath(".coldquanta"),
```

Comment:
we probably only need to look in the `.qiskit` directory

Reply:
assuming this came from `gss.superstaq_client` - the first one checks the XDG data directory (e.g. `~/.local/share/super.tech`), second one looks for a hidden folder in the home directory (`~/.super.tech`)